gobblin-core/src/test/java/gobblin/source/extractor/extract/AbstractSourceTest.java

```diff
@@ -0,0 +1,235 @@
+package gobblin.source.extractor.extract;
+
+import com.google.common.collect.Lists;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.SourceState;
+import gobblin.configuration.State;
+import gobblin.configuration.WorkUnitState;
+import gobblin.configuration.WorkUnitState.WorkingState;
+import gobblin.source.extractor.Extractor;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class AbstractSourceTest {
+
+  WorkUnitState committedWorkUnitState;
+  WorkUnitState successfulWorkUnitState;
+  WorkUnitState failedWorkUnitState;
+  List<WorkUnitState> previousWorkUnitStates;
+
+  @BeforeClass
+  public void setUpBeforeClass() {
+    WorkUnitState committedWorkUnitState = new WorkUnitState();
+    committedWorkUnitState.setWorkingState(WorkingState.COMMITTED);
+    WorkUnitState successfulWorkUnitState = new WorkUnitState();
+    successfulWorkUnitState.setWorkingState(WorkingState.SUCCESSFUL);
+    WorkUnitState failedWorkUnitState = new WorkUnitState();
+    failedWorkUnitState.setWorkingState(WorkingState.FAILED);
+    previousWorkUnitStates = Lists.newArrayList();
```

Comment:
You can do previousWorkUnitStates = Lists.newArrayList(this. committedWorkUnitState, this. successfulWorkUnitState, this. failedWorkUnitState); Also please use this. in front of all references of member variables.